### PR TITLE
Update to vcf_results to support vcf.gz files

### PR DESCRIPTION
**Pull request by jonk-ebi**
### Description
Update to vcf_results.get_results_from_path to support vcf.gz files

### Related JIRA Issue(s)
[ENSWBSITES-2704](https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-2704)

### Review App URL(s) 
N/A

### Knowledge Base
This update also resolves a regression introduced during the restructuring of the repo. Unit tests for vcf will now run

### Example(s)
An easy way to test this change is to remove the skip from `test_get_results_with_file_and_dump` and update the `vcf_path` to a local file. This will load in the file and create a json dump of the expected page. Test expects a full page of 100 variants for page 2 so if your test file does not include >300 variants you will need to change this `assert`

### Checklist

- [x] Black formatting
- [x] Tests

### Dependencies 

If you get any errors about headers like `length` make sure you are using `git+https://github.com/likhitha-surapaneni/vcfpy@header-fix` for vcfpy

### CODE REVIEW

File: app/vep/utils/vcf_results.py [hunk 1]
```diff
@@ -229,10 +239,8 @@ def get_results_from_stream(
...
         metadata=model.Metadata(
             pagination=model.PaginationMetadata(
-                page=page,
-                per_page=page_size,
-                total=total
+                page=page, per_page=page_size, total=total
             )
         ),
-        variants=variants
+        variants=variants,
```
Comment (jyothishnt) [inline app/vep/utils/vcf_results.py:245]:
```suggestion
        variants=variants
```

File: app/vep/utils/vcf_results.py [hunk 2]
```diff
@@ -1,10 +1,11 @@
 """ Module for loading a VCF and parsing it into a VepResultsResponse
 object as defined in APISpecification"""
 
+from typing import List, Dict, Any, IO
+
 import vcfpy
-from typing import List, Dict, Any
 
-from vep.models import vcf_results_model as model
+from app.vep.models import vcf_results_model as model
```
Comment (jyothishnt) [inline app/vep/utils/vcf_results.py:8]: do you need `app.`? complained for me locally.

Comment (jyothishnt) [reply app/vep/utils/vcf_results.py:8]: Aah! yes to run the test you need to access it using `app.`  but not for the app.

File: app/vep/utils/vcf_results.py (hunk 1 above)
Comment (jonk-ebi) [reply app/vep/utils/vcf_results.py:245]: Black adds the trailing `,`. Python is happy with it and it is believed that it is nicer for anyone extending the code to add additional elements

File: app/vep/utils/vcf_results.py (hunk 2 above)
Comment (jonk-ebi) [reply app/vep/utils/vcf_results.py:8]: Yeah I needed that and IO before the unit tests would run

**Pull request closed without merging**